### PR TITLE
Add/Remove channel actions

### PR DESCRIPTION
**Pull request by maksim-karatkevich**

### CODE REVIEW

File: src/app/features/Chanels/redux/actions.js [hunk 1]
```diff
@@ -1,24 +1,35 @@
...
 
 export function fetchChannels() {
   return async (dispatch, getState) => {
     const channelsExists = getState().channelsState.length > 0;
-
     if (!channelsExists) {
-      requestChannels().then((result) => {
-        dispatch(getChannels(result));
-      });
+      const channels = localStorage.getItem('channelsList');
```
Comment (Roma36) [inline src/app/features/Chanels/redux/actions.js:24]: Please, encapslate localStorage logic in **requestChannels()** function.

The end result should be something like this:
```
if (!channelsExists) {
    const channels = await requestChannels();
    dispatch(getChannelsAction(channels))
}
```

File: src/app/features/Chanels/redux/actions.js [hunk 2]
```diff
@@ -33,9 +44,45 @@ export function fetchPosts(id) {
       requestPosts(id).then((result) => {
         const postItem = result.find((item) => item.channelId === id);
         if (postItem) {
-          dispatch(getPosts(postItem));
+          dispatch(getPostsAction(postItem));
```
Comment (Roma36) [inline src/app/features/Chanels/redux/actions.js:47]: please, return one post from **requestPosts** function. As soon as you provide channel id to this function, you don't have to find the posts in action creator.

File: src/app/features/Chanels/redux/actions.js [hunk 3]
```diff
@@ -33,9 +44,45 @@ export function fetchPosts(id) {
...
 }
+
+export function addChannel(channelName) {
+  return async (dispatch, getState) => {
+    const channelExists = getState().channelsState.find(
+      (channel) => channel.name === channelName
+    );
+
+    if (!channelExists) {
+      requestChannel(channelName).then((result) => {
```
Comment (Roma36) [inline src/app/features/Chanels/redux/actions.js:61]: please, rename this function to **createChannel** and encapsulate localStorage logic in it. Do NOT mix localStorage logic and redux logic.

Comment (Roma36) [reply src/app/features/Chanels/redux/actions.js:61]: the **createChannel** should return newly created channel item.

File: src/app/redux/alertAction.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  type: SHOW_ALERT,
+  payload: alert,
+});
+
+function showAlert(alertState) {
+  return async (dispatch) => {
+    dispatch(showAlertAction(alertState));
+    setTimeout(() => {
+      dispatch(
+        showAlertAction({
```
Comment (Roma36) [inline src/app/redux/alertAction.js:13]: add another action for hiding an alert. You can name it **hideAlert** or **resetAlertState**. In reducer, you'll have to just return the initialState of alert redcer.

File: src/app/features/Chanels/components/AddChannelForm.js [hunk 5]
```diff
@@ -1,14 +1,49 @@
...
+import PropTypes from 'prop-types';
 
-import { Button, Input } from 'antd';
+const AddChannelForm = ({ onSubmitForm }) => {
+  const [form] = Form.useForm();
+  const [, forceUpdate] = useState();
+
+  // To disable submit button at the beginning.
+  useEffect(() => {
+    forceUpdate({});
```
[the remark this answers is not in the document]
Comment (maksim-karatkevich) [reply src/app/features/Chanels/components/AddChannelForm.js:11]: Just copied from https://ant.design/components/form/?locale=en-US#header 
removed

**Pull request merged**